### PR TITLE
docs(api.md): expand on puppeteer-core vs puppeteer

### PR DESCRIPTION
**Pull request by aslushnikov**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -307,6 +308,29 @@ The Puppeteer API is hierarchical and mirrors the browser structure.
...
+Chromium, which it then drives using `puppeteer-core`. Being an end-user product, `puppeteer` supports a bunch of convenient `PUPPETEER_*` env variables to tweak its behavior.
+
+`puppetee-core` is a *library* to help drive anything that supports devtools protocol. `puppeteer-core` doesn't download Chromium when installed. Being a library, `puppeteer-core` is fully driven
+through its programmatic interface and disregards all the `PUPPETEER_*` env variables.
+
+To sum up, the only differences between `puppeteer-core` and `puppeteer` are:
+- `puppeteer-core` doesn't automatically download Chromium when installed.
+- `puppeteer-core` ignores all `PUPPETEER_*` env variables.
+
+In most cases, you'll be fine using the `puppeteer` package right away.
```
Comment (ebidel) [inline docs/api.md]: remove "right away"

File: docs/api.md [hunk 2]
```diff
@@ -307,6 +308,29 @@ The Puppeteer API is hierarchical and mirrors the browser structure.
...
+### puppeteer vs puppeteer-core
+
+Every release since v1.7.0 we publish two packages:
+- [puppeteer](https://www.npmjs.com/package/puppeteer)
+- [puppeteer-core](https://www.npmjs.com/package/puppeteer-core)
+
+`puppeteer` is a *product* for browser automation. When installed, it downloads a version of
+Chromium, which it then drives using `puppeteer-core`. Being an end-user product, `puppeteer` supports a bunch of convenient `PUPPETEER_*` env variables to tweak its behavior.
+
+`puppetee-core` is a *library* to help drive anything that supports devtools protocol. `puppeteer-core` doesn't download Chromium when installed. Being a library, `puppeteer-core` is fully driven
```
Comment (ebidel) [inline docs/api.md]: devtools => DevTools

File: docs/api.md [hunk 3]
```diff
@@ -307,6 +308,29 @@ The Puppeteer API is hierarchical and mirrors the browser structure.
...
+through its programmatic interface and disregards all the `PUPPETEER_*` env variables.
+
+To sum up, the only differences between `puppeteer-core` and `puppeteer` are:
+- `puppeteer-core` doesn't automatically download Chromium when installed.
+- `puppeteer-core` ignores all `PUPPETEER_*` env variables.
+
+In most cases, you'll be fine using the `puppeteer` package right away.
+
+However, you should use `puppeteer-core` if:
+- you're building another end-user product or library atop of DevTools protocol. For example, one might build PDF generator using `puppeteer-core` and write a custom `install.js` script that downloads [`headless_shell`](https://chromium.googlesource.com/chromium/src/+/lkgr/headless/README.md) instead of Chromium to save up space.
```
Comment (ebidel) [inline docs/api.md]: save up space -> save disk space.

File: docs/api.md [hunk 4]
```diff
@@ -307,6 +308,29 @@ The Puppeteer API is hierarchical and mirrors the browser structure.
...
+- you're bundling puppeteer to use in Chrome Extension / Browser context. `puppeteer-core` contains all the code necessary to work with DevTools protocol and doesn't download unnecessary binary.
```
Comment (ebidel) [inline docs/api.md]: puppeteer -> Puppeteer

Comment (ebidel) [inline docs/api.md]: with the DevTools protocol where downloading an additional Chromium binary is unnecessary.

**Review by ebidel — CHANGES_REQUESTED** [review]

**Review by ebidel — APPROVED** [review]

File: docs/api.md (hunk 1 above)
Comment (aslushnikov) [reply docs/api.md]: Done.

File: docs/api.md (hunk 2 above)
Comment (aslushnikov) [reply docs/api.md]: Done.

File: docs/api.md (hunk 4 above)
Comment (aslushnikov) [reply docs/api.md]: Done.

Comment (aslushnikov) [reply docs/api.md]: Done.

File: docs/api.md (hunk 3 above)
Comment (aslushnikov) [reply docs/api.md]: Done.

File: README.md [hunk 5]
```diff
@@ -51,6 +51,8 @@ npm i puppeteer-core
 
 `puppeteer-core` is intended to be a lightweight version of puppeteer for launching an existing browser installation or for connecting to a remote one.
 
+See [puppeteer vs puppeteer-core](https://github.com/GoogleChrome/puppeteer/blob/v1.7.0/docs/api.md#puppeteer-vs-puppeteer-core).
```
Comment (vsemozhetbyt) [inline README.md:54]: @aslushnikov It seems this link does not work.

Comment (aslushnikov) [reply README.md:54]: indeed; addressed in https://github.com/GoogleChrome/puppeteer/pull/3197.

**Pull request merged**